src/heretic/r_plane.c

```diff
@@ -515,6 +516,8 @@ void R_DrawPlanes(void)
 
         tempSource = W_CacheLumpNum(lumpnum, PU_STATIC);
 
+        if (flattranslation[pl->picnum] != -1) // [crispy] adapt swirl from src/doom to src/heretic
```

Comment:
Sorry, but I told you *in detail* which two lines would need to get changed in order to get this merged. As it is now, if `flattranslation[pl->picnum] == -1` then `lumpnum = firstflat - 1` and you call `W_ReleaseLumpNum(lumpnum)` some 20 lines later.

I acknowledge the effort that's 28 commits, but I can't merge it like this.